test/e2e/upgrade.go

```diff
@@ -0,0 +1,166 @@
+	err = os.Chmod(clusterctlBinaryPath, 0744) //nolint:gosec
+	Expect(err).ToNot(HaveOccurred(), "failed to chmod temporary file")
+
+	By("Initializing the workload cluster with older versions of providers")
+
+	// NOTE: by default we are considering all the providers, no matter of the contract.
+	// However, given that we want to test both v1alpha3 --> v1beta1 and v1alpha4 --> v1beta1, the INIT_WITH_PROVIDERS_CONTRACT
+	// variable can be used to select versions with a specific contract.
+	contract := "v1alpha4"
+
+	clusterctl.InitManagementClusterAndWatchControllerLogs(ctx, clusterctl.InitManagementClusterAndWatchControllerLogsInput{
+		ClusterctlBinaryPath:    clusterctlBinaryPath, // use older version of clusterctl to init the management cluster
+		ClusterProxy:            targetCluster,
+		ClusterctlConfigPath:    clusterctlConfigPath,
+		CoreProvider:            e2eConfig.GetProviderLatestVersionsByContract(contract, config.ClusterAPIProviderName)[0],
+		BootstrapProviders:      e2eConfig.GetProviderLatestVersionsByContract(contract, config.KubeadmBootstrapProviderName),
+		ControlPlaneProviders:   e2eConfig.GetProviderLatestVersionsByContract(contract, config.KubeadmControlPlaneProviderName),
+		InfrastructureProviders: e2eConfig.GetProviderLatestVersionsByContract(contract, e2eConfig.InfrastructureProviders()...),
+		LogFolder:               filepath.Join(artifactFolder, "clusters", cluster.Name),
+	}, e2eConfig.GetIntervals(specName, "wait-controllers")...)
+
+	By("Configure Ironic Configmap")
+	configureIronicConfigmap(true)
+
+	By("Add labels to BMO CRDs")
+	labelBMOCRDs(nil)
+
+	By("Install BMO")
+	installIronicBMO(targetCluster, "false", "true")
+
+	By("Install Ironic in the target cluster")
+	installIronicBMO(targetCluster, "true", "false")
+
+	By("Restore original BMO configmap")
+	restoreBMOConfigmap()
+	By("Reinstate Ironic Configmap")
+	configureIronicConfigmap(false)
+
+	By("Add labels to BMO CRDs in the target cluster")
+	labelBMOCRDs(targetCluster)
+
+	By("Ensure API servers are stable before doing move")
+	Consistently(func() error {
+		kubeSystem := &corev1.Namespace{}
+		return bootstrapClusterProxy.GetClient().Get(ctx, client.ObjectKey{Name: "kube-system"}, kubeSystem)
+	}, "1s", "100ms").Should(BeNil(), "Failed to assert bootstrap API server stability")
+	Consistently(func() error {
+		kubeSystem := &corev1.Namespace{}
+		return targetCluster.GetClient().Get(ctx, client.ObjectKey{Name: "kube-system"}, kubeSystem)
+	}, "1s", "100ms").Should(BeNil(), "Failed to assert target API server stability")
+
+	By("Get the management cluster images before upgrading")
+	pods, err := targetClusterClientSet.CoreV1().Pods("").List(ctx, metav1.ListOptions{})
+	printImages(pods)
+
+	By("Upgrading providers to the latest version available")
+	Logf("clusterv1.GroupVersion.Version: %v", clusterv1.GroupVersion.Version)
+	Logf("os.Getenv: %v", os.Getenv("CONFIG_FILE_PATH"))
+	CONFIG_FILE_PATH := os.Getenv("HOME")+"/.cluster-api/clusterctl.yaml"
```

Comment:
If you stay with this method of configuring the config path then I think this should look something like:
`configFilePath := os.Getenv("HOME")+"/.cluster-api/clusterctl.yaml"
Logf("os.Getenv: %v", configFilePath)`